westcoast/src/main/java/frc/commands/Autonomous.java

```diff
@@ -23,22 +25,23 @@ public static Command move(@NotNull Definition definition, double distance) {
     }
 
     public static Command shootAndLeave(@NotNull Definition definition) {
-        return shoot(definition, () -> 3100.0)
+        return shoot(definition, () -> 3100.0, () -> Shoot.Mode.Full)
                 .andThen(move(definition, 3.0));
     }
 
-    public static Command fullAuto(@NotNull Definition definition, IntakeCollect.ColorSupplier color) {
-        return shoot(definition, () -> 3100.0)
+    public static Command fullAuto(
+            @NotNull Definition definition,
+            @NotNull BooleanSupplier ballAcceptanceStrategy
+    ) {
+        return shoot(definition, () -> 3100.0, () -> Shoot.Mode.Full)
                 .alongWith(deploy(definition))
                 .andThen(
                         move(definition, 2.0)
-                                .alongWith(
-                                        collect(definition, color)
-                                )
+                                .alongWith(collect(definition, ballAcceptanceStrategy))
                                 .until(definition.subsystems.magazine::hasFeederSensorBreak)
                 )
                 .andThen(move(definition, -2.0))
-                .andThen(shoot(definition, () -> 3100.0))
+                .andThen(shoot(definition, () -> 3100.0, () -> Shoot.Mode.Full))
```

Comment:
Should be Mode.Single, not Mode.Full because it only has a single preload at the start of a match

Reply:
it keeps the original behavior and does not make a difference for the application.

Reply:
Fair, in theory if we have the option to single shot, i think it makes sense to single shot it to lower time